src/hotspot/share/runtime/flags/jvmFlagConstraintsRuntime.cpp

```diff
@@ -32,6 +32,21 @@
 #include "runtime/task.hpp"
 #include "utilities/powerOfTwo.hpp"
 
+JVMFlag::Error AOTModeConstraintFunc(ccstr value, bool verbose) {
+  if (strcmp(value, "off") != 0 &&
+      strcmp(value, "record") != 0 &&
+      strcmp(value, "create") != 0 &&
+      strcmp(value, "auto") != 0 &&
+      strcmp(value, "on")) {
```

Comment:
This should be strcmp(value, "on") != 0 ??

Reply:
Okay both ways are valid, you could also remove the other "!= 0", the mixing was confusing

Reply:
Hotspot style rule is "no implicit booleans" so the check for `!= 0` should be used in all cases.

Reply:
Fixed.